service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -645,7 +645,7 @@ Description=Configure environment for terra
 After=jupyter.service
 
 [Service]
-ExecStart=${TERRA_BOOT_SCRIPT}
+ExecStart=-${TERRA_BOOT_SCRIPT}
```

Comment:
My concern with this is that the TERRA_BOOT_SCRIPT could error out for many reasons, including syntax errors and other trivial bugs. We'd want to catch that and it would be better if we had some hard failures as a result. This doesn't necessarily get us there, but ignoring the exit code I think will be another mask.

Does `terra resource mount` have fine grained exit codes? It would be nice if it did map some errors to things like:

https://opensource.apple.com/source/xnu/xnu-201/bsd/sys/errno.h.auto.html

```
#define	ENOENT		2		/* No such file or directory */
...
#define	EACCES		13		/* Permission denied */
```

Then the TERRA_BOOT_SCRIPT could at least handle known things:
```
# Mount terra workspace resources
"${USER_HOME_LOCAL_BIN}/terra" resource mount
readonly MOUNT_RC="$?"

case "${MOUNT_RC}" in 
  # If a user doesn't have permissions or the bucket doesn't exist, don't error out here
  0) # EOK
  2) # ENOENT
  13) # EACCESS
    >&2 echo "Ignoring error from 'terra resource mount'"
    exit 0
    ;;
  *)
    >&2 echo "Unexpected error '${MOUNT_RC}' from 'terra resource mount'"
    exit 1
esac
```

What do you think?

Reply:
Appreciate the quick review! Agree that this is the more correct approach. It would be good to add these return codes to single resource mounts. For mounting multiple resources, we aggregate the list of mount errors into a single return code. There aren't fine grained exit codes, It's essentially `1 if sum(exitCodes) > 1 else 0`. It's not clear to me how to best bubble up mount errors with varying causes. We could modify the behavior to only return a nonzero exit code if an _**unexpected**_ error occurs. Would that be a good intermediary solution?

Reply:
that seems reasonable to me fwiw

Reply:
Made changes so that the boot service no longer ignores non zero exit codes from the boot script itself.

Errors in the script itself will prevent the service from running. Non zero exit codes from commands executed inside the boot script will no longer cause the service to fail, but instead be logged to `.terra/boot-output.txt` for debugging.

We don't want errors from child commands to break the service since they don't necessarily occur each vm startup and can likely be resolved by the user (in the case of mounting bad referenced buckets).

PTAL @wnojopra @amygdala

Reply:
This seems like a reasonable approach to me— if it later turns out that we want to bubble up some of the child errors differently, we can iterate.